ibm/resource_ibm_ipsec_vpn.go

```diff
@@ -0,0 +1,466 @@
+		Delete:   resourceIBMIPSecVPNDelete,
+		Update:   resourceIBMIPSecVPNUpdate,
+		Exists:   resourceIBMIPSecVPNExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"datacenter": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"internal_peer_ip_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"phase_one": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"authentication": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateAuthProtocol,
+						},
+						"encryption": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateEncyptionProtocol,
+						},
+						"diffie_hellman_group": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validateDiffieHellmanGroup,
+						},
+						"keylife": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validatekeylife,
+						},
+					},
+				},
+			},
+			"phase_two": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"authentication": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "MD5",
+							ValidateFunc: validateAuthProtocol,
+						},
+						"encryption": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "3DES",
+							ValidateFunc: validateEncyptionProtocol,
+						},
+						"diffie_hellman_group": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      2,
+							ValidateFunc: validateDiffieHellmanGroup,
+						},
+						"keylife": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      3600,
+							ValidateFunc: validatekeylife,
+						},
+					},
+				},
+			},
+			"address_translation": { //Parameters for creating an adress translation
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"remote_ip_adress": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"internal_ip_adress": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"notes": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"preshared_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"customer_peer_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"internal_subnet_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"remote_subnet_id": { //customer subnet id . need atleast one customer subnet id for applying the configuratons
```

Comment:
As per description user can either provide remote_subnet_id or remote_subnet so add ConflictsWith: []string{"remote_subnet"}, respectively to both attributes..

Reply:
Done